temporal-sdk/src/main/java/io/temporal/internal/worker/WorkflowWorker.java

```diff
@@ -98,13 +98,13 @@ public WorkflowWorker(
     this.pollerOptions = getPollerOptions(options);
     this.workerMetricsScope =
         MetricsTag.tagged(options.getMetricsScope(), WorkerMetricsTag.WorkerType.WORKFLOW_WORKER);
+    slotSupplier.setMetricsScope(workerMetricsScope);
```

Comment:
Could we not take a more generic slot supplier in the constructor of `WorkflowWorker` and construct the `TrackingSlotSupplier` here where we have the metric scope.

Reply:
Yeah probably, let me check that out